cmake/Platform/Mac/runtime_dependencies_mac.cmake.in

```diff
@@ -212,14 +212,23 @@ if(@target_file_dir@ MATCHES ".app/Contents/MacOS")
             file(REMOVE_RECURSE ${remove_file_list})
         endif()
 
-
     endif()
 
+    cmake_path(SET exec_path "${bundle_path}")
+
 else() # Non-bundle case
 
     if(depends_on_python)
         # RPATH fix python
         execute_process(COMMAND ${LY_INSTALL_NAME_TOOL} -change @rpath/Python @rpath/Python.framework/Versions/Current/Python @target_file@)
     endif()
 
+    cmake_path(SET exec_path "@target_file@")
+
 endif()
+
+cmake_path(GET exec_path FILENAME runtime_output_filename)
+cmake_path(SET entitlement_file "$ENV{TARGET_TEMP_DIR}/${runtime_output_filename}.xcent")
+foreach(config @CMAKE_CONFIGURATION_TYPES@)
+    configure_file(@LY_ROOT_FOLDER@/cmake/Platform/Mac/runtime_install_mac.cmake.in @CMAKE_BINARY_DIR@/runtime_install/${config}/$ENV{TARGET_NAME}.cmake @ONLY)
```

Comment:
After looking into this, it's not straight-forward to try to compute `$ENV{TARGET_TEMP_DIR}`. For eg, for AssetProcessor target, it evaluates to `/Users/<user>/O3DE/o3de/build/mac/Code/Tools/AssetProcessor/O3DE.build/profile/AssetProcessor.build/` and for a gem it would evaluate to `/Users/<user>/O3DE/o3de/build/mac/External/GameState-6b72fb8a/Code/O3DE.build/profile/GameState.build`. Depending on whether the target is gem or not, we'd have to splice together an output directory, relative path from source or gem directory, project name, target name, config. That seems more inconvenient/complicated than having to do an explicit configure/build which is also only required when hardened runtime is enabled.